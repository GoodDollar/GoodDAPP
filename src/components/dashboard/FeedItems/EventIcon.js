import React from 'react'
import { Icon } from '../../common'
import { withStyles } from '../../../lib/styles'
import getEventSettingsByType from './EventSettingsByType'

const EventIcon = ({ type, theme, styles, style, size = 34 }) => {
  const icon = getEventSettingsByType(theme, type)
  return <Icon color={icon.color} size={size} name={icon.name} style={[styles.eventIcon, style ? style : {}]} />
}

const getStylesFromProps = ({ theme }) => ({
  eventIcon: {
    marginRight: 0,
  },
})

export default withStyles(getStylesFromProps)(EventIcon)
