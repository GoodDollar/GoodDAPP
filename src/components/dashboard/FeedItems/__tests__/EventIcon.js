import React from 'react'
import renderer from 'react-test-renderer'
import EventIcon from '../EventIcon'

// Note: test renderer must be required after react-native.

describe('EventIcon - Withdraw', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<EventIcon type="withdraw" />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<EventIcon type="withdraw" />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('EventIcon - Send', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<EventIcon type="send" />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<EventIcon type="send" />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('EventIcon - Receive', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<EventIcon type="receive" />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<EventIcon type="receive" />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('EventIcon - welcome', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<EventIcon type="welcome" />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<EventIcon type="welcome" />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
