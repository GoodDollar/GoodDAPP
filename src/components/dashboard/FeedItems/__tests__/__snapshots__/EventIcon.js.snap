// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventIcon - Receive matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
/>
`;

exports[`EventIcon - Send matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
/>
`;

exports[`EventIcon - Withdraw matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
/>
`;

exports[`EventIcon - welcome matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
/>
`;
